cmd/kubeadm/app/constants/constants.go

```diff
@@ -538,6 +540,42 @@ func GetDNSIP(svcSubnet string) (net.IP, error) {
 	return dnsIP, nil
 }
 
+// GetKubernetesServiceCIDR returns the default Service CIDR for the Kubernetes internal service
+func GetKubernetesServiceCIDR(svcSubnetList string, isDualStack bool) (*net.IPNet, error) {
+	if isDualStack {
+		// The default address family for your cluster is the address family of the first
+		// service cluster IP range configured via the `--service-cluster-ip-range` flag
+		// to the kube-controller-manager.
+		svcCIDRs, err := utilnet.ParseCIDRs(strings.Split(svcSubnetList, ","))
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to parse ServiceSubnet %v", svcSubnetList)
+		}
+		if len(svcCIDRs) == 0 {
+			return nil, errors.New("no Service CIDR(s) specified")
+		}
+		return svcCIDRs[0], nil
+	}
+	// internal IP address for the API server
+	_, svcSubnet, err := net.ParseCIDR(svcSubnetList)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error parsing CIDR %q", svcSubnetList)
```

Comment:
we probably should have this error the same as above?
```
return nil, errors.Wrapf(err, "unable to parse ServiceSubnet %v", svcSubnetList)
```

Reply:
done